### DISCUSSION TITLE
Question about using hook-delete-policy on Argo Workflow resource

### DISCUSSION THREAD

**Discussion by masterginger** [argoproj/argo-cd#16468] [Q&A]
(Moved from https://github.com/argoproj/argo-workflows/discussions/12280#discussion-5901929)

Hi all,

We were looking at [hook deletion policy](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/#hook-deletion-policies), (which mentioned it could be applied to Workflow resources) and tried to add the following annotations to the Workflow resource:
```
argocd.argoproj.io/hook: PostSync
argocd.argoproj.io/hook-delete-policy: HookSucceeded
```
However, we observed that after the Workflow resource was created and synced, it was deleted before the workflow itself actually completed. There were still pods (for workflow steps) running. As a result, the workflow execution was interrupted (the pods for workflow steps were deleted along with the workflow resource itself) and most of the workflow steps were never run.

Is this the expected behavior? or adding hook delete policy to Workflow resource is not a valid use case in the first place?

Any feedback is appreciated!

**Comment by coder-pie** [comment]
Running into the same issue. Can we get some answers please?

**Comment by ChristianCiach** [comment]
There was a similar issue that has been fixed ages ago:

* https://github.com/argoproj/argo-cd/issues/6541

The Gitops-engine bundles a health-check for Workflow resources, but maybe it broke again? See https://github.com/argoproj/gitops-engine/blob/master/pkg/health/health_argo.go#L29

The health check assumes the workflow to be in "progressing" state when the `.status.phase` attribute is empty or equal to "Running" or "Pending". Can you please manually check this attribute of the Workflow while it is running? If the attribute is not in one of these three states, we can probably workaround this issue by providing a [custom health check](https://argo-cd.readthedocs.io/en/stable/operator-manual/health/#custom-health-checks) (until it gets fixed properly).

**Reply by coder-pie** [reply]
Thank you for the response :) the status indeed changed to progressing and the hook did execute before it got deleted! The issue was with the script it was executing, and it all happened to fast and was working before I changed the hook deletion policy which erroneously led me to believe that it's an issue with the policy.  Appreciate your help on this :)
